kie-dmn-validation/src/main/java/org/kie/dmn/validation/DMNValidatorImpl.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kie.dmn.validation;
+
+import static java.util.stream.Collectors.*;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import javax.swing.tree.DefaultMutableTreeNode;
+import javax.swing.tree.TreeNode;
+import javax.xml.XMLConstants;
+import javax.xml.transform.Source;
+import javax.xml.transform.stream.StreamSource;
+import javax.xml.validation.Schema;
+import javax.xml.validation.SchemaFactory;
+
+import org.drools.core.util.IoUtils;
+import org.kie.api.KieServices;
+import org.kie.api.builder.KieBuilder;
+import org.kie.api.builder.KieFileSystem;
+import org.kie.api.builder.Message;
+import org.kie.api.builder.Message.Level;
+import org.kie.api.builder.ReleaseId;
+import org.kie.api.builder.Results;
+import org.kie.api.event.rule.DefaultRuleRuntimeEventListener;
+import org.kie.api.event.rule.ObjectInsertedEvent;
+import org.kie.api.runtime.KieContainer;
+import org.kie.api.runtime.StatelessKieSession;
+import org.kie.dmn.feel.model.v1_1.DMNModelInstrumentedBase;
+import org.kie.dmn.feel.model.v1_1.Definitions;
+import org.kie.dmn.validation.ValidationMsg.Severity;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.xml.sax.SAXException;
+
+public class DMNValidatorImpl implements DMNValidator {
+    public static Logger LOG = LoggerFactory.getLogger(DMNValidatorImpl.class);
+    static Schema schema;
+    static {
+        try {
+            schema = SchemaFactory.newInstance(XMLConstants.W3C_XML_SCHEMA_NS_URI).newSchema();
+        } catch (SAXException e) {
+            e.printStackTrace();
+        }
+    }
+    
+    /**
+     * A KieContainer is normally available,
+     * unless at runtime some problem prevented building it correctly.
+     */
+    private Optional<KieContainer> kieContainer;
+    /**
+     * Collect at init time the runtime issues which prevented to build the `kieContainer` correctly.
+     */
+    private List<ValidationMsg> failedInitMsg;
+    
+    public DMNValidatorImpl() {
+        KieServices ks = KieServices.Factory.get();
```

Comment:
the constructor perform the creation of the "embedded" KieContainer as discussed, and the potential issue on creation (although they should not happen) are captured.